File: app/src/main/java/art/luaj/hyperisle/ui/fragment/mainnav/settings/SettingsFragment.java
package art.luaj.hyperisle.ui.fragment.mainnav.settings;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.preference.PreferenceFragmentCompat;

import art.luaj.hyperisle.R;
import art.luaj.hyperisle.databinding.FragmentSettingsBinding;
import art.luaj.hyperisle.ext.BaseFragment;

public class SettingsFragment extends BaseFragment<FragmentSettingsBinding> {
    @Override
    protected FragmentSettingsBinding inflate(LayoutInflater inflater, ViewGroup container, boolean attachToRoot) {
        return FragmentSettingsBinding.inflate(inflater, container, attachToRoot);
    }

    @Override
    protected void initFragment(FragmentSettingsBinding binding) {
    }

    public static class PreferenceFragment extends PreferenceFragmentCompat {
        @Override
        public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
            setPreferencesFromResource(R.xml.prefs_settings, rootKey);
        }
    }
}

File: app/src/main/java/art/luaj/hyperisle/ui/fragment/SettingsFragment.kt
package art.luaj.hyperisle.ui.fragment

import android.os.Bundle
import androidx.lifecycle.ViewModel
import androidx.preference.PreferenceFragmentCompat
import art.luaj.hyperisle.R
import art.luaj.hyperisle.databinding.FragmentSettingsBinding
import art.luaj.hyperisle.ext.Tools
import art.luaj.hyperisle.ui.base.BaseFragment

class SettingsFragment : BaseFragment<FragmentSettingsBinding, ViewModel>(FragmentSettingsBinding::inflate, null) {
    override fun initFragment(binding: FragmentSettingsBinding, viewModel: ViewModel?) {
        childFragmentManager.beginTransaction().add(R.id.setting_container, PreferenceFragment()).commitNow()
    }

    class PreferenceFragment : PreferenceFragmentCompat() {

        public fun getKey(Text: CharSequence): CharSequence? {
            return Tools.concat("system_preference_", Text);
        }

        override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
            addPreferencesFromResource(R.xml.prefs_settings);
            /*
            val hideIcon = findPreference(getKey("hide_icon"));
            if (hideIcon != null) {
                hideIcon.setOnPreferenceClickListener {
                    // 处理点击事件
                    true
                }
            }
             */
        }

    }
}